src/engine/renderer/tr_surface.cpp

```diff
@@ -1183,14 +1183,14 @@ static void Tess_SurfaceMD5( md5Surface_t *srf )
 				VectorMA( binormal, *boneWeight, tmp, binormal );
 			}
 
-			VectorNormalize( normal );
-			VectorNormalize( tangent );
-			VectorNormalize( binormal );
+			VectorNormalizeFast( normal );
+			VectorNormalizeFast( tangent );
+			VectorNormalizeFast( binormal );
 			VectorCopy( position, tessVertex->xyz );
 
 			R_TBNtoQtangents( tangent, binormal, normal, tessVertex->qtangents );
 
-			floatToHalf( surfaceVertex->texCoordsF, tessVertex->texCoords );
```

Comment:
@slipher would you know why this code didn't aborted compilation wile the function is expecting vec4 but the arguments are vec2?

Reply:
In C any array of T type is considered equivalent to a pointer to T when used in a function argument. This is why `vec3_t` acts like a reference when used in a function argument.
All of these function declarations are equivalent:
```
void foo(int* v);
void foo(int v[]);
using int4 = int[4]; void foo(int4 v);
void foo(int v[6]);
```
If you really want to force the inputs to be declared as arrays of that exact length you can make the signature
`void floatToHalf( const float(&floats)[4], f16_t (&halfs)[4] )`